graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/FilterOperator.java

```diff
@@ -0,0 +1,268 @@
+    }
+
+    @Override
+    protected Relation buildRelation(ColumnRelationBuilder<Relation> relationStart, Term key) {
+      return relationStart.containsKey(key);
+    }
+
+    @Override
+    public Relation buildRelation(Column column, Object value, NameMapping nameMapping) {
+      return buildRelation(
+          Relation.column(CqlIdentifier.fromInternal(column.name())),
+          toCqlKeyTerm(column, value, nameMapping));
+    }
+  },
+  CONTAINS_ENTRY("containsEntry") {
+    @Override
+    protected Condition buildCondition(ConditionBuilder<Condition> conditionBuilder, Term value) {
+      throw new IllegalStateException("CONTAINS ENTRY can't be used on IF conditions");
+    }
+
+    @Override
+    protected Relation buildRelation(ColumnRelationBuilder<Relation> relationStart, Term value) {
+      throw new IllegalStateException("CONTAINS ENTRY requires key and value terms");
+    }
+
+    @Override
+    public Relation buildRelation(Column column, Object value, NameMapping nameMapping) {
+      Column.ColumnType mapType = column.type();
+      assert mapType != null && mapType.isMap();
+      Map<String, Object> entry = (Map<String, Object>) value;
+      Column.ColumnType keyType = mapType.parameters().get(0);
+      Term keyTerm = DataTypeMapping.toCqlTerm(keyType, entry.get("key"), nameMapping);
+      Column.ColumnType valueType = mapType.parameters().get(1);
+      Term valueTerm = DataTypeMapping.toCqlTerm(valueType, entry.get("value"), nameMapping);
+      return Relation.mapValue(CqlIdentifier.fromInternal(column.name()), keyTerm)
+          .isEqualTo(valueTerm);
+    }
+  },
+  ;
+
+  private static final Map<String, FilterOperator> mapByFieldName = buildMapByFieldName();
+  private final String fieldName;
+
+  public static FilterOperator fromFieldName(String fieldName) {
+    FilterOperator op = mapByFieldName.get(fieldName);
+    if (op == null) {
+      throw new IllegalArgumentException("Invalid filter field name: " + fieldName);
+    }
+    return op;
+  }
+
+  private static Map<String, FilterOperator> buildMapByFieldName() {
+    return Arrays.stream(FilterOperator.values())
+        .collect(Collectors.toMap(FilterOperator::getFieldName, o -> o));
+  }
+
+  public Condition buildCondition(Column column, Object value, NameMapping nameMapping) {
+    return buildCondition(
+        Condition.column(CqlIdentifier.fromInternal(column.name())),
+        DataTypeMapping.toCqlTerm(column.type(), value, nameMapping));
+  }
+
+  protected abstract Condition buildCondition(
+      ConditionBuilder<Condition> conditionBuilder, Term value);
+
+  public Relation buildRelation(Column column, Object value, NameMapping nameMapping) {
+    return buildRelation(
+        Relation.column(CqlIdentifier.fromInternal(column.name())),
+        DataTypeMapping.toCqlTerm(column.type(), value, nameMapping));
+  }
+
+  protected abstract Relation buildRelation(
+      ColumnRelationBuilder<Relation> relationStart, Term value);
```

Comment:
Nit: that abstract method doesn't feel right, only some of the children use it, yet the others still have to implement it and throw. It makes the code a bit harder to follow.

I think a static utility method would be a better way to factor the common parts of simple operators:
```java
  EQUAL("eq") {
     ...
    @Override
    public Relation buildRelation(Column column, Object value, NameMapping nameMapping) {
      return buildSimpleRelation(column, value, nameMapping, ArithmeticRelationBuilder::isEqualTo);
    }
  },

  private static Relation buildSimpleRelation(
      Column column,
      Object value,
      NameMapping nameMapping,
      BiFunction<ColumnRelationBuilder<Relation>, Term, Relation> operator) {
    return operator.apply(
        Relation.column(CqlIdentifier.fromInternal(column.name())),
        DataTypeMapping.toCqlTerm(column.type(), value, nameMapping));
  }
```

A similar thing can probably be done for conditions too, since they use the same approach.

Reply:
Good point, I've refactored it as suggested.